frontend/packages/ceph-storage-plugin/src/components/kms-config/vault-config.tsx

```diff
@@ -28,92 +64,165 @@ export const ValutConfigure: React.FC<KMSConfigureProps> = ({
 }) => {
   const { t } = useTranslation();
 
-  const kms: VaultConfig = state.kms?.[ProviderNames.VAULT] || state.kms;
-  const kmsObj: VaultConfig = _.cloneDeep(kms);
+  // webhooks
+  const isKmsVaultSASupported = useFlag(GUARDED_FEATURES.OCS_KMS_VAULT_SERVICE_ACCOUNT);
+
+  const vaultState: VaultConfig = state.kms?.[ProviderNames.VAULT] || state.kms;
+  const vaultStateClone: VaultConfig = _.cloneDeep(vaultState);
+  const { encryption } = state;
+
+  const updateVaultState = (vaultConfig: VaultConfig) => {
+    mode
+      ? setEncryptionDispatch(ActionType.SET_KMS_ENCRYPTION, mode, dispatch, vaultConfig)
+      : dispatch({
+          type: 'securityAndNetwork/setVault',
+          payload: vaultConfig,
+        });
+  };
+
+  const setAuthMethod = (authMethod: VaultAuthMethods) => {
+    vaultStateClone.authMethod = authMethod;
+    updateVaultState(vaultStateClone);
+  };
+
+  const filteredVaultAuthMethodMapping = Object.values(VaultAuthMethodMapping).filter(
+    (authMethod) =>
+      (encryption.clusterWide
+        ? authMethod.supportedEncryptionType.includes(KmsEncryptionLevel.CLUSTER_WIDE)
+        : true) &&
+      (encryption.storageClass
+        ? authMethod.supportedEncryptionType.includes(KmsEncryptionLevel.STORAGE_CLASS)
+        : true),
+  );
```

Comment:
So, basically if user has selected both the check-boxes (clusterWide & storageClass) we will not show the `Service Account` method in the dropdown. This is intentional, right ?

Reply:
Yes, that is correct